net: promiscuous: Fix crash in promiscuous mode

icsys-aal opened the pull request:
When a packet can't be cloned we crash as we try to initialize the cursor on a nullptr. We should check if we have a valid pointer, and if we don't we drop the packet along with a warning.

jukkar reviewed (commented):
The change is good, thanks for fixing it. Just fix the compliance issues and we are ready to merge this.

icsys-aal commented:
Fixed the code formatting, let me know if there's anything else you need me to change.

jukkar commented on subsys/net/ip/net_if.c:
```diff
@@ -4145,12 +4145,16 @@ enum net_verdict net_if_recv_data(struct net_if *iface, struct net_pkt *pkt)
+			}
+		} else {
+			NET_WARN("packet dropped, unable to clone packet");
+			verdict = NET_DROP;
```
After revisiting this change, I am not sure about this `NET_DROP`. The reason being that the function should return info whether the original net_pkt should be dropped or not, and in this case the earlier call
`verdict = net_if_l2(iface)->recv(iface, pkt);` returns that information.
So for the cloned pkt, imho we should not set the verdict here. To me it looks like that if we return `NET_DROP` here, the original pkt is dropped without any reason. WDYT?

jukkar replied:
Also in the case of the cloning failed, perhaps better error msg would be
`NET_WARN("promiscuous packet dropped, unable to clone packet");`
in order to tell the reader what exactly happened.

icsys-aal replied:
After looking into it I agree with your suggestion and have removed the verdict part.

jukkar commented:
There is still mismatch for your signed-off-by, please see
https://github.com/zephyrproject-rtos/zephyr/actions/runs/6183457730/job/16787172023?pr=62582

jukkar commented:
@icsys-aal please check the compliance issues, the checker is very picky about your commit author name, it should match the signed-off-by

icsys-aal committed:
net: promiscuous: Fix crash in promiscuous mode

When a packet can't be cloned we crash as we try to initialize the
cursor on a nullptr. We should check if we have a valid pointer,
and if we don't we drop the packet along with a warning.

Signed-off-by: Andreas Ålgård <aal@ixys.no>
